watch-library/shared/watch/watch_private_display.c

```diff
@@ -64,14 +64,15 @@ void watch_display_character(uint8_t character, uint8_t position) {
     } else {
         if (character == 'R') character = 'r'; // R needs to be lowercase almost everywhere
     }
-    if (position == 0) {
-        watch_clear_pixel(0, 15); // clear funky ninth segment
-    } else {
+    if (position != 0) {
```

Comment:
A bit confused here - why was this pixel cleared but not the other special pixels?

Reply:
There are 8 bits in a byte. This means we can map up to 8 segments per digit "automagically" using the [character set](https://github.com/joeycastillo/Sensor-Watch/blob/5f0ebd8c66b3512d67efc8174c5f65a86488a4ea/watch-library/shared/watch/watch_private_display.h#L30) and [segment map](https://github.com/joeycastillo/Sensor-Watch/blob/5f0ebd8c66b3512d67efc8174c5f65a86488a4ea/watch-library/shared/watch/watch_private_display.h#L129) data. Most digits have only seven segments, but two have more than seven: the weekday digits in position 0 and 1. This makes the eighth bit useful for controlling an extra segment, where available. In position 0, we use the eighth bit to control the center downstroke of the W or M. In position 2, we use that eighth bit to control the extra segment at the top left that creates the `T` in **ST**opwatch. 

Writing any character to these two positions automagically sets and clears the "extra" eighth segment, because they are controlled by that eighth bit. Alas, position 0 has a _ninth_ segment at the bottom left, and we can't automagically control it with the segment mapping code because there aren't nine bits in a byte. So we manually clean it up; it's the only segment in the text display that we can't control with the eight bits per character we have available.

Reply:
Yup, I think I got this, I was just confused bywatch_set_pixel(0, 12) and watch_set_pixel(1, 12) below since they didn't have corresponding clears. But having now looked at https://joeycastillo.github.io/Sensor-Watch-Documentation/segmap.html, I can see that these are quite... strange segments :) Thanks!